cmd/tink-worker/action.go

```diff
@@ -43,15 +43,13 @@ func executeAction(ctx context.Context, action *pb.WorkflowAction, wfID string)
 		return pb.ActionState_ACTION_IN_PROGRESS, errors.Wrap(err, "DOCKER CREATE")
 	}
 	l.With("containerID", id, "command", action.GetOnTimeout()).Info("container created")
-
-	var timeCtx context.Context
-	var cancel context.CancelFunc
+	// Setting time context for action
+	timeCtx := ctx
 	if action.Timeout > 0 {
+		var cancel context.CancelFunc
 		timeCtx, cancel = context.WithTimeout(context.Background(), time.Duration(action.Timeout)*time.Second)
```

Comment:
First of all, the global timeout will be controlled from the `tink-server` which is on the provisioner machine so we won't be passing the global timeout here. 
Second if a user has provided a timeout for a particular action I would like to give that action a separate context with that deadine so that in future we can run actions(with timeout) in parllel. That's why I have not used parent context here. But yes if user has not provided the timeout for an action it should run in a parent context which is as  you mentioned `ctx`.

Reply:
Agree with @detiber. If (when, I have a proposal to write for this today) we have concurrent actions we can avoid the parent context, but for here/now lets continue with using the parent.

Reply:
Sure, for now I can change it to parent context for now and in future when we have concurrent actions this needs to be changed again.

Reply:
Done